Enable multi-model

tgerdesnv opened the pull request:
Enable multi-model (--run-config-profile-models-concurrently-enable)
Added L0_multi_model_profile test
Updated PA get_cmd() for multi-model to use MPI
Updated auto_adjust_parameters() support to handle multi-model

tgerdesnv commented on model_analyzer/perf_analyzer/perf_analyzer.py:
```diff
@@ -263,8 +304,11 @@ def _poll_perf_analyzer(self, process):
-        Use of the perf analyzer process
+        Attempt to update PA parameters based on the output
         """
+        # TODO-TMA-518 - how to handle?
```
@matthewkotila I need to figure out how to handle this auto-adjusting for multi-model.

tgerdesnv replied:
fixed

tgerdesnv commented on model_analyzer/record/metrics_manager.py:
```diff
@@ -439,7 +439,7 @@ def _run_perf_analyzer(self, run_config, perf_output_writer):
 
         if perf_output_writer:
-            # TODO-TMA-518: MPI command
+            # TODO-TMA-518: MPI command - still need to fix
```
Random printing outside of PA needs to be cleaned up

tgerdesnv replied:
fixed

tgerdesnv commented on model_analyzer/perf_analyzer/perf_analyzer.py:
```diff
@@ -104,7 +104,7 @@ def __init__(self, path, config, max_retries, timeout, max_cpu_util):
         self._max_cpu_util = max_cpu_util
 
-        # TODO-TMA-518: Need to update for multi-model
+        # TODO-TMA-518: Need to update for multi-model - still need to fix
```
fixed

nv-braf commented on qa/L0_multi_model_profile/check_results.py:
```diff
@@ -0,0 +1,101 @@
+        - X is the number of measurements that we would expected 
+        for model 1 in a single-model case
+        - Y is the number of measurements that we would expected 
+        for model 1 in a single-model case
```
this should be model 2 (or you can change this to model A/B which is my preference: models get letters, configs get numbers)

tgerdesnv replied:
fixed

nv-braf reviewed (changes requested):
I've observed a crash with my local run - we need to debug before this is released

tgerdesnv commented:
> I've observed a crash with my local run - we need to debug before this is released

Fixed! I also fixed two other issues -- one a potential issue around unsorted batch size list

nv-braf reviewed (approved):
Looks good!
